### PR TITLE
[issue 113] - Refactor deployments, tools and remove legacy name identifiers

### LINKED ISSUE #113: Refactor (rename symbols, packages, move files etc.) to prepare for first release
- [x] Move `deployments` within `testsuite`
- [x] Move `core` and `provisioners` under parent POM
- [x] Refactor (rename) the WildFly Legacy Bootable JAR provisioning tooling and tests in order to relate to EAP 7, as it has been done for the EAP 7 legacy s2i (#106) and template (#104) cases.   

* Environment: deployments,tools/intersmash-tooling-provisioners,testsuite
* Steps to reproduce: N/A

### PR DESCRIPTION
**Pull request by fabiobrz**
## Description
- Refactoring deployments, they are testsuite stuff, so moved there
- Refactoring tools, now `core` and `provisioners` are top level
- As for the EAP 7 Hello World example, the legacy WildFly Bootable JAR deployment should only be built and tested for the product, i.e. EAP 7, and not being used to verify the newer WildFly versions.
- Fixed, removed some wrong, although working, configurations

Fix #113 

**Note for reviewer(s)**
A lot of files have been changed but don't be scared, this PR is basically and mostly about renaming files. Just a few minor changes  have been added to those, but here's a per-commit recap to make your work quicker:

[**1.**](https://github.com/Intersmash/intersmash/pull/129/commits/43b6e7a7aea264b514fc8dcd2908aaeb539864d6) Mainly renaming files. 
A POM configuration has been added since it was missing in the EAP logic so far. Code cleaning.
[**2.**](https://github.com/Intersmash/intersmash/pull/129/commits/e7ecd4c8678308a9194efc208b0f75e43045590e) Fixing WIldFLy/EAP XP BOMs definitions
[**3.**](https://github.com/Intersmash/intersmash/pull/129/commits/7d716cb154507df4a6ab4ac3e8e43e31f99cd081) BIG move
Basically moved files + consequent package name refactoring affecting imports + CI update
testsuite modles split in deployments + integration-tests.
Renamed demos to examples as well (sorry)
BTW broken commit containing an unresolved conflict
[**4.**](https://github.com/Intersmash/intersmash/pull/129/commits/be29530b082a72195579bc6afb9bffdaad3f315f) Just fixing examples modules imports
[**5.**](https://github.com/Intersmash/intersmash/pull/129/commits/a7bdc14171fa8c183a1ca52c4ead6cd76d2da868) BIG move
Basically moved files + consequent package name refactoring affecting imports + CI update
[**6.**](https://github.com/Intersmash/intersmash/pull/129/commits/8a7e32964df733433c32beb7088b4fe97075264f) Fixing the Bootable JAR plugin configuration after moving core and provisioners up, and removing a redundant lombok dep
[**7.**](https://github.com/Intersmash/intersmash/pull/129/commits/18039416a6db0d2d97c31a97ee335c51d86cfaf9) Fixing the CI scripts again and a SPI registration file containing a wrong package
[**8.**](https://github.com/Intersmash/intersmash/pull/129/commits/01616b0a43822fcc731131eee1c10cac6b086b9d) Renaming the profile for configuring the excluded tests depending on TS execution target (community/prod)
Renaming the profile for configuring the excluded tests depending on wildfly/eap target version (eap/eapxp)
[**9.**](https://github.com/Intersmash/intersmash/pull/129/commits/00ac6eef3c774629fc7f728cf96d97ba53c3ed55) the e2e scripts need to set `intersmash.deployments.repository.url`and `intersmash.deployments.repository.ref` to those of the PR, otherwise deployments are cloned from the Intersmash repo itself, and this doesn't work when modifying a deployment.

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [x] Breaking change (fix or feature that would cause existing functionality to change **(but we're not breaking APIs since we haven't released yet)**
 - [x] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [x] Pull Request contains a description of the changes
 - [x] Pull Request does not include fixes for multiple issues/topics
 - [x] Code is self-descriptive and/or documented
 - [x] I have implemented unit tests to cover my changes
 - [x] I tested my code in OpenShift

### CODE REVIEW

File: testsuite/integration-tests/pom.xml [hunk 1]
```diff
@@ -61,48 +65,43 @@
...
-                        <configuration combine.children="append">
-                            <excludedGroups>${intersmash.test.execution.excludedGroups}</excludedGroups>
-                        </configuration>
-                    </plugin>
-                </plugins>
-            </build>
         </profile>
         <profile>
-            <id>ts.community</id>
+            <id>ts.execution-profile.community</id>
```
Comment (marekkopecky) [inline testsuite/integration-tests/pom.xml:90]: What is the reason for having this profile? My understanding is that this provide just overrides default values with the same values. Can we remove this profile? Or remove definitions of intersmash.test.execution.profile and intersmash.test.excludedGroups.by.execution-profile from defaults and enabling this profile automatically once ts.wildfly.target-distribution.eap won't be enabled?

Comment (fabiobrz) [reply testsuite/integration-tests/pom.xml:90]: +1, removing it and leaving defaults.

File: deployments/intersmash-deployments-provider/src/main/java/org/jboss/intersmash/deployments/IntersmashSharedDeployments.java [hunk 2]
```diff
@@ -28,16 +28,16 @@
  * applications. They need to be installed in local repository.
  */
 public class IntersmashSharedDeployments {
-	static final String BOOTABLE_JAR_ARTIFACT = "wildfly-bootable-jar";
-	static final String BOOTABLE_JAR_ARTIFACT_JAVAX = "wildfly-bootable-jar-javax";
+	static final String WILDFLY_BOOTABLE_JAR = "wildfly-bootable-jar";
+	static final String EAP_7_BOOTABLE_JAR = "eap7-bootable-jar";
```
Comment (marekkopecky) [inline deployments/intersmash-deployments-provider/src/main/java/org/jboss/intersmash/deployments/IntersmashSharedDeployments.java]: Where is this property used?

Comment (marekkopecky) [reply deployments/intersmash-deployments-provider/src/main/java/org/jboss/intersmash/deployments/IntersmashSharedDeployments.java]: Hmm, sorry, it seems that I'm blind, closing.

**Comment by marekkopecky** [comment]
AFAICS `<version.wildfly-jar-maven-plugin>` property is defined in 4 places to 10.0.0.Final and on 1 place to 6.1.3.Final-redhat-00001. Would be possible to define one default value to 10.0.0.Final and just replace it once to 6.1.3.Final-redhat-00001?

File: testsuite/deployments/pom.xml [hunk 3]
```diff
@@ -101,4 +143,5 @@
             </build>
         </profile>
     </profiles>
-</project>
+
+</project>
```
Comment (marekkopecky) [inline testsuite/deployments/pom.xml:147]: Will you please add missing new line here?

File: testsuite/integration-tests/pom.xml [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+            <id>ts.wildfly.target-distribution.eap</id>
+            <properties>
+                <!-- Exclude tests that shouldn't be executed when dealing base EAP, i.e. Bootable Jar tests
+                (Bootable JAR is only available with WildFLy or EAP XP, ot base EAP) -->
+                <intersmash.test.excludedGroups.by.wildfly-target-distribution>ts.wildfly.requires-bootable-jar</intersmash.test.excludedGroups.by.wildfly-target-distribution>
+            </properties>
+        </profile>
+    </profiles>
+
+</project>
```
Comment (marekkopecky) [inline testsuite/integration-tests/pom.xml:108]: same here

**Comment by marekkopecky** [comment]
I go through the current topics and just find a few minor issues. Please also let me know once CI would be fixed or share some details about the test errors. Thank you for this MR!

**Comment by fabiobrz** [comment]
> AFAICS `<version.wildfly-jar-maven-plugin>` property is defined in 4 places to 10.0.0.Final and on 1 place to 6.1.3.Final-redhat-00001. Would be possible to define one default value to 10.0.0.Final and just replace it once to 6.1.3.Final-redhat-00001?

Thanks @marekkopecky - I removed one of them which was redundant. The others are meant to be there for the following reasons:
1. examples are basically not depending on one another, hence each of them could use a different version
2. the root POM should not have such a property which is only relevant in the testsuite or examples contexts

File: testsuite/integration-tests/pom.xml (hunk 4 above)
Comment (fabiobrz) [reply testsuite/integration-tests/pom.xml]: Done, thanks.

File: testsuite/deployments/pom.xml (hunk 3 above)
Comment (fabiobrz) [reply testsuite/deployments/pom.xml]: Fixed, thanks

**Comment by fabiobrz** [comment]
> I go through the current commits and just find a few minor issues. Please also let me know once CI would be fixed or share some details about the test errors. Thank you for this MR!

Thanks a lot for your review @marekkopecky - let's see whether the last run fixes the only failure (which I was not able to reproduce at first attempt locally), otherwise I'll try and work on it it later or on Monday.

**Comment by marekkopecky** [comment]
> 1. examples are basically not depending on one another, hence each of them could use a different version

Yes, this is true. Theoretically we could probably have one example for WF and second one for EAP, so there might be a need for this. This said, the versions are the same now in both examples. Let's keep it as it is, but if we find out that there won't be any need for different versions, we can move the version property to examples/pom.xml

**Comment by fabiobrz** [comment]
> I go through the current commits and just find a few minor issues. Please also let me know once CI would be fixed or share some details about the test errors. Thank you for this MR!

Ok @marekkopecky - I fixed the last tests and CI checks are green now. Let me know whether you think there's something else we'd want to do here.

**Comment by marekkopecky** [comment]
Thank you! PR merged.

**Pull request merged**